src/vm/nativelibrary.cpp

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+// 
+// File: NativeLibrary.cpp
+//
+
+//
+// Unmanaged implementation of the NativeLibrary class
+//
+
+
+#include "common.h"
+#include "nativelibrary.h"
+#include "dllimport.h"
+
+//
+// GetProcAddress
+// Returns the address of the specified symbol from the target module, or nullptr if not found
+LPVOID QCALLTYPE NativeLibrary::GetProcAddress(HMODULE hModule, LPCSTR lpProcName)
+{
+    QCALL_CONTRACT;
+
+    // Arguments are check on managed side
+    PRECONDITION(hModule != nullptr);
+
+    LPVOID procAddress = nullptr;
+
+    BEGIN_QCALL;
+
+    procAddress = ::GetProcAddress(hModule, lpProcName);
+
+    END_QCALL;
+
+    return procAddress;
+}
+
+//
+// LoadLibrary
+// Loads the specified library, returning its handles, or nullptr if library cannot be found
+HINSTANCE QCALLTYPE NativeLibrary::LoadLibrary(LPCUTF8 moduleName, QCall::AssemblyHandle callingAssembly, BOOL searchAssemblyDirectory, DWORD searchPaths)
+{
```

Comment:
is it possible that expose these functions(overload is fine) into dllimport.cpp, so that all Pinvoke related functions are implemented in dllimport.cpp?

Reply:
All of these functions are implemented in dllimport.cpp adjacent to the existing code. This is just the QCALL entry point with parameter validation.